docs/dev/guide/l2-l1.md

```diff
@@ -1,3 +1,42 @@
 # L2 -> L1 communication
 
-Sending messages from zkSync to Ethereum is not yet available, but it will be implemented later.
+This section describes the interface for interaction with Ethereum from L2. It assumes that you are already familiar with the basic concepts of working with the L2 -> L1 communication. If you are new to this topic, you can read the conceptual introduction [here](../zksync-v2/l1-l2-interop.md). 
+
+## Structure
+
+Unlike the L1 -> L2 communication, it is impossible to directly initialize transactions from L2 to L1. However, you can send an arbitrary-length message from zkSync to Ethereum and then handle the received message on L1 smart contract. From the L2 side, to send a message special system contract should be called. It accepts only bytes of the message that will be sent to the Ethereum `zkSync` smart contract. From the L1 side, the `zkSync` smart contract provides an interface to prove that message was sent to L1 in a certain zkSync block. 
+
+## Sending a message on L2
+
+::: sender of the message will be determined from context (i.e. `msg.sender`)
+
+```
+function sendToL1(bytes memory _message) external returns (bytes32 messageHash);
+```
+
+- `_message` is a parameter that contains raw bytes of the message
+
+The function sends a message from L2 and returns the keccak256 hash of the message bytes. The hash of the message can be used later for getting proof that the message was sent on L1. But its use is optional and is for convenience purposes only.
+
+## Prove inclusion of the message into the L2 block
+
+```
+    struct L2Message {
+        address sender;
+        bytes data;
+    }
+
+    function proveL2MessageInclusion(
+        uint32 _blockNumber,
+        uint256 _index,
+        L2Message calldata _message,
+        bytes32[] calldata _proof
+    ) external view returns (bool);
+```
+
+- `_blockNumber` is a parameter that points to the **L2** block number in which the message was sent.
+- `_index` is a parameter that contains the serial number of the message in L2 block. Can be restored both from observing Ethereum and received via API.
+- `_message` is a parameter that contains full information of the sent message, including raw bytes of payload and sender address.
+- `_proof` is a parameter that contains Merkle proof of the message inclusion. Can be restored both from observing Ethereum and received via API.
+
+The function returns a boolean value whose value indicates the success of proving that message with such parameters was sent to L1.
```

Comment:
Once we add the information to the SDK about obtaining the proofs from the server, we should add the link to it here (cc @ly0va)

Reply:
Why don't we have this sentence at the top?

The sentence is hard to read -- why do we need "proving"? can't we just say "indicates that a message... was sent to L1"?

```suggestion
This function returns a boolean value that indicates the success of proving that a message, with such parameters, was sent to L1.

Reply:
> Why don't we have this sentence at the top?

I use text from L1 -> L2 communication (https://github.com/matter-labs/zksync-web-v2-docs/blob/main/docs/dev/guide/l1-l2.md?plain=1#L58), leave the same format, but change sense of the sentences.